bentoml/utils/s3.py

```diff
@@ -33,54 +38,78 @@ def is_s3_url(url):
         return False
 
 
-def upload_to_s3(s3_url, file_path):
-    """
-    Update files in the file_path to the s3 location
-    """
+def upload_directory_to_s3(
+    upload_directory_path, region, bucket_name, s3_path_prefix=''
+):
+    s3_client = boto3.client('s3', region)
+    try:
+        for root, _, file_names in os.walk(upload_directory_path):
+            relative_path_to_upload_dir = os.path.relpath(root, upload_directory_path)
+            if relative_path_to_upload_dir == '.':
+                relative_path_to_upload_dir = ''
+            for file_name in file_names:
+                key = os.path.join(
+                    s3_path_prefix, relative_path_to_upload_dir, file_name
+                )
+                logger.debug(
+                    'Uploading {name} to s3 {location}'.format(
+                        name=file_name, location=bucket_name + '/' + key
+                    )
+                )
+                s3_client.upload_file(os.path.join(root, file_name), bucket_name, key)
+    except Exception as error:
+        raise BentoMLException(str(error))
+
+
+def create_s3_bucket_if_not_exists(bucket_name, region):
+    s3_client = boto3.client('s3', region)
+    try:
+        s3_client.get_bucket_acl(Bucket=bucket_name)
+        logger.debug('Use existing s3 bucket')
+    except ClientError as error:
+        if error.response and error.response['Error']['Code'] == 'NoSuchBucket':
+            logger.debug('Creating s3 bucket: {}'.format(bucket_name))
```

Comment:
I believe linter will warn about this..

Reply:
W1202: Use % formatting in logging functions and pass the % parameters as arguments (logging-format-interpolation)

Reply:
also would be nice to add the region parameter to the log message